### DISCUSSION TITLE
is there a way to get hermit to install a specific version thats not in the hermit-packages?

### DISCUSSION THREAD

**Discussion by jstrachan** [cashapp/hermit#63] [General]
do we need to create a PR on the https://github.com/cashapp/hermit-packages repository for every release of every package; or is there a way to be able to install the latest stable version created after the last PR to https://github.com/cashapp/hermit-packages 

I thought the @stable channel was meant to refresh every 24 hours but I've not been able to figure out with any permutation of `hermit upgrade/sync/uninstall/install with $package-$version or $package-@stable or jx@3` a way to install any other version than the exact 2 versions in the hermit-packages.

Is there a trick to find other versions to install?

e.g. I'm trying to install any version of `jx` after the last version in the hermit-packages of `3.2.140` via https://github.com/cashapp/hermit-packages/blob/master/jx.hcl#L12 and so far I can only get the versions from the package: https://github.com/cashapp/hermit-packages/blob/master/jx.hcl#L12

FWIW we release `jx` quite frequently - we could automate PRs on the https://github.com/cashapp/hermit-packages  but I didn't want to overload you if there's a way to use a channel instead

**Comment by alecthomas** [comment]
Channels can retrieve from two sources currently.

1. A stable URL without a `${version}`. See eg. [rust nightly](https://github.com/cashapp/hermit-packages/blob/master/rust.hcl#L16).
2. A version pattern match. eg. [jx](https://github.com/cashapp/hermit-packages/blob/master/jx.hcl#L14).

The former will periodically check the URL for updates while the latter will only check versions contained within the manifest, meaning the manifest itself will have to be updated.

For Hermit itself we publish each release to a stable URL, eg. https://github.com/cashapp/hermit/releases/download/stable/hermit-darwin-amd64.gz

So yep those are currently your two choices.

We do have plans to include automatic updates for GitHub releases. We'll have a cron job that runs in the hermit-packages repo, that checks for new releases of all packages hosted on GitHub, and updates the manifest programatically.

But for now I think just send a PR, that's fine.

**Comment by alecthomas** [comment]
@jstrachan I've implemented this [here](https://github.com/cashapp/hermit/pull/69). I'll tag a new release with this support, then after that release rolls out (24h) you can modify the jx manifest like so:

```hcl
version "3.2.137" "3.2.140" {
  auto-version {
    github-release = "jenkins-x/jx"
  }
}
```

Then I'll add some support to the hermit-packages repo to automatically apply the auto-version command.

**Reply by alecthomas** [reply]
cc @juliaogris who was also interested in this

**Reply by alecthomas** [reply]
I've added the workflow to [hermit-packages](https://github.com/cashapp/hermit-packages/actions/workflows/autoversion.yml), so this should be fully automatable now.

**Comment by alecthomas** [comment]
This is now [working](https://github.com/cashapp/hermit-packages/compare/1450efdad414...a71301d55214)!
